features/ubuntu_pro.feature

```diff
@@ -659,3 +659,48 @@ Feature: Command behaviour when auto-attached in an ubuntu PRO image
            | bionic  |
            | focal   |
            | jammy   |
+
+    @series.xenial
+    @series.jammy
+    @uses.config.machine_type.gcp.pro
+    Scenario Outline: Unregistered Pro machine
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        contract_url: 'https://contracts.canonical.com'
+        data_dir: /var/lib/ubuntu-advantage
+        log_level: debug
+        log_file: /var/log/ubuntu-advantage.log
+        """
+        And I create the file `/tmp/response-overlay.json` with the following:
+        """
+        {
+            "https://contracts.canonical.com/v1/clouds/gcp/token": [
+            {
+              "code": 400,
+              "type": "contract",
+              "response": {
+                "code": "contract not found",
+                "message": "missing product mapping for subscription \\"subscription\\", plan \\"plan\\", product \\"product\\", publisher \\"canonical\\", sku \\"sku\\" ",
+                "traceId": "id"
+              }
+            }]
+        }
+        """
+        And I append the following on uaclient config:
+        """
+
```

Comment:
@orndorffgrant @renanrodrigo @CalvoM when reviewing this PR, please let me know if you can spot any problems by removing this extra line here

Reply:
I can confirm that the test fails without this newline with the following output:
```
2022-08-24 15:16:28,927:DEBUG:pycloudlib.instance:executing: sudo pro auto-attach
2022-08-24 15:16:29,324:DEBUG:root:stdout: 
2022-08-24 15:16:29,324:DEBUG:root:stderr: ERROR:root:Unhandled exception, please file a bug
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/uaclient/cli.py", line 1759, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/uaclient/cli.py", line 1835, in main
    cfg = config.UAConfig(root_mode=is_root)
  File "/usr/lib/python3/dist-packages/uaclient/config.py", line 124, in __init__
    self.cfg, self.invalid_keys = parse_config(self.cfg_path)
  File "/usr/lib/python3/dist-packages/uaclient/config.py", line 659, in parse_config
    cfg.update(yaml.safe_load(system.load_file(config_path)))
  File "/usr/lib/python3/dist-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/usr/lib/python3/dist-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/usr/lib/python3/dist-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "/usr/lib/python3/dist-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/lib/python3/dist-packages/yaml/parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
  File "/usr/lib/python3/dist-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/usr/lib/python3/dist-packages/yaml/scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
  File "/usr/lib/python3/dist-packages/yaml/scanner.py", line 577, in fetch_value
    raise ScannerError(None, None,
yaml.scanner.ScannerError: mapping values are not allowed here
  in "<unicode string>", line 4, column 48:
     ... log/ubuntu-advantage.logfeatures:
                                         ^
Unexpected error(s) occurred.
For more details, see the log: /var/log/ubuntu-advantage.log
To file a bug run: ubuntu-bug ubuntu-advantage-tools
```

Reply:
The steps that create files is not adding a newline since 60591d92179720f6646c6939b5637d412ba7961b 
So when modifying the file, you need this newline here, or at the end of the step which creates the config file.